addons/website/controllers/form.py

```diff
@@ -31,18 +31,29 @@ def website_form(self, model_name, **kwargs):
         csrf_token = request.params.pop('csrf_token', None)
         if request.session.uid and not request.validate_csrf(csrf_token):
             raise BadRequest('Session expired (invalid CSRF token)')
-
         try:
             # The except clause below should not let what has been done inside
             # here be committed. It should not either roll back everything in
             # this controller method. Instead, we use a savepoint to roll back
             # what has been done inside the try clause.
             with request.env.cr.savepoint():
                 if request.env['ir.http']._verify_request_recaptcha_token('website_form'):
-                    return self._handle_website_form(model_name, **kwargs)
-            error = _("Suspicious activity detected by Google reCaptcha.")
+                    response = self._handle_website_form(model_name, **kwargs)
+                else:
+                    raise UserError(_("Suspicious activity detected by Google reCaptcha."))
```

Comment:
I'd avoid adding new error throwing
```suggestion
                    error = _("Suspicious activity detected by Google reCaptcha.")
                    response = json.dumps({'error': error})
```